core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -101,100 +103,134 @@ type EthBroadcaster struct {
 	ethTxInsertListener pg.Subscription
 	eventBroadcaster    pg.EventBroadcaster
 
-	ks        txmgrtypes.KeyStore[gethCommon.Address, *big.Int, int64]
-	addresses []gethCommon.Address
+	ks               txmgrtypes.KeyStore[ADDR, *big.Int, int64]
+	enabledAddresses []ADDR
 
-	checkerFactory TransmitCheckerFactory
+	checkerFactory TransmitCheckerFactory[ADDR, TX_HASH]
 
 	// triggers allow other goroutines to force EthBroadcaster to rescan the
 	// database early (before the next poll interval)
 	// Each key has its own trigger
-	triggers map[gethCommon.Address]chan struct{}
+	triggers map[string]chan struct{}
 
 	chStop chan struct{}
 	wg     sync.WaitGroup
 
+	initSync  sync.Mutex
+	isStarted bool
 	utils.StartStopOnce
 }
 
 // NewEthBroadcaster returns a new concrete EthBroadcaster
-func NewEthBroadcaster(orm ORM, ethClient evmclient.Client, config Config,
-	keystore txmgrtypes.KeyStore[gethCommon.Address, *big.Int, int64],
+func NewEthBroadcaster(
+	txStorageService EvmTxStorageService,
+	ethClient evmclient.Client,
+	config Config,
+	keystore EvmKeyStore,
 	eventBroadcaster pg.EventBroadcaster,
-	addresses []gethCommon.Address, resumeCallback ResumeCallback,
-	txAttemptBuilder txmgrtypes.TxAttemptBuilder[*evmtypes.Head, gas.EvmFee, gethCommon.Address, gethCommon.Hash, EthTx, EthTxAttempt],
-	logger logger.Logger, checkerFactory TransmitCheckerFactory, autoSyncNonce bool) *EthBroadcaster {
+	enabledAddresses []*evmtypes.Address,
+	txAttemptBuilder EvmTxAttemptBuilder,
+	nonceSyncer EvmNonceSyncer,
+	logger logger.Logger,
+	checkerFactory EvmTransmitCheckerFactory,
+	autoSyncNonce bool,
+) *EvmEthBroadcaster {
 
-	triggers := make(map[gethCommon.Address]chan struct{})
 	logger = logger.Named("EthBroadcaster")
-	return &EthBroadcaster{
+	return &EvmEthBroadcaster{
 		logger:           logger,
-		orm:              orm,
+		txStorageService: txStorageService,
 		ethClient:        ethClient,
 		TxAttemptBuilder: txAttemptBuilder,
-		resumeCallback:   resumeCallback,
```

Comment:
Can't provide it at constructor time, since we don't have it. We only get it when a Txmgr client sets it on Txmgr via RegisterResumeCallback(). Check that function.